service/controller/resource/instance/create_master_instances_upgrading.go

```diff
@@ -336,6 +343,13 @@ func (r *Resource) updateInstance(ctx context.Context, customObject providerv1al
 
 	r.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("ensured instance '%s' to be updated", instanceName))
 
+	go func() {
+		err := r.startInstanceWatchdog(ctx, g, s)
+		if err != nil {
+			r.logger.LogCtx(ctx, "level", "error", "message", fmt.Sprintf("Watchdog failed for instance '%s'", instanceName))
+		}
+	}()
+
```

Comment:
this feel like ugly hack, you ahve no control over this and it will be hard to din how is this executed or how its terminated, can have some leaking goroutines as well

Reply:
this IS an ugly hack. If anybody has better ideas I'm more than happy to get rid of it.

Reply:
@calvix for the records, we're working on a better version of this hack to make it safer.
But this won't be a blocker for the release.
We will deploy it if we complete it by the end of the day, otherwise it'll go online as is.